core/test/utils/batch.hpp

```diff
@@ -92,6 +117,68 @@ std::unique_ptr<MatrixType> generate_uniform_batch_random_matrix(
 }
 
 
+/**
+ * Generate a batch of 1D Poisson matrices on the reference executor.
+ *
+ * @param exec  The reference executor.
+ * @param nrows  The size (number of rows) of the generated matrix
+ * @param nbatch  The number of Poisson matrices in the batch
+ */
+template <typename ValueType, typename IndexType = int>
+std::unique_ptr<matrix::BatchCsr<ValueType, IndexType>> create_poisson1d_batch(
+    std::shared_ptr<const ReferenceExecutor> exec, const int nrows,
+    const size_type nbatch)
+{
+    const int nnz = 3 * (nrows - 2) + 4;
+    gko::Array<IndexType> row_ptrs(exec, nrows + 1);
+    {
+        const auto ra = row_ptrs.get_data();
+        ra[0] = 0;
+        ra[1] = 2;
+        for (int i = 2; i < nrows; i++) {
+            ra[i] = ra[i - 1] + 3;
+        }
+        ra[nrows] = ra[nrows - 1] + 2;
+        GKO_ASSERT(ra[nrows] == nnz);
+    }
+    gko::Array<IndexType> col_idxs(exec, nnz);
+    {
+        const auto ca = col_idxs.get_data();
+        ca[0] = 0;
+        ca[1] = 1;
+        for (int i = 1; i < nrows - 1; i++) {
+            const int rstart = row_ptrs.get_const_data()[i];
+            ca[rstart] = i - 1;
+            ca[rstart + 1] = i;
+            ca[rstart + 2] = i + 1;
+        }
+        const auto lrstart = row_ptrs.get_const_data()[nrows - 1];
+        ca[lrstart] = nrows - 2;
+        ca[lrstart + 1] = nrows - 1;
+    }
+    gko::Array<ValueType> vals(exec, nnz * nbatch);
+    for (int i = 0; i < nbatch; i++) {
+        ValueType *const va = vals.get_data() + i * nnz;
+        va[0] = 2.0;
+        va[1] = -1.0;
+        for (int i = 1; i < nrows - 1; i++) {
+            const int rstart = row_ptrs.get_const_data()[i];
+            va[rstart] = -1.0;
+            va[rstart + 1] = 2.0;
+            va[rstart + 2] = -1.0;
+        }
+        const auto lrstart = row_ptrs.get_const_data()[nrows - 1];
+        va[lrstart] = -1.0;
+        va[lrstart + 1] = 2.0;
+    }
```

Comment:
I think you can simplify this with a `matrix_assembly_data`, where you can create triplets `(row_idx, col_idx, value)` depending on the index. https://github.com/ginkgo-project/ginkgo/blob/develop/include/ginkgo/core/base/matrix_assembly_data.hpp#L119